packages/browser/src/transports/base.ts

```diff
@@ -54,6 +65,63 @@ export abstract class BaseTransport implements Transport {
     return this._buffer.drain(timeout);
   }
 
+  /**
+   * @inheritDoc
+   */
+  public recordLostEvent(type: Outcome, category: SentryRequestType): void {
+    if (!this.options.sendClientReport) {
+      return;
+    }
+    const key = `${type}:${CATEGORY_MAPPING[category]}`;
+    logger.log(`Adding ${key} outcome`);
+    this._outcomes[key] = (this._outcomes[key] ?? 0) + 1;
+  }
+
+  /**
+   * Send outcomes as an envelope
+   */
+  protected _flushOutcomes(): void {
+    if (!this.options.sendClientReport) {
+      return;
+    }
+
+    if (!navigator || typeof navigator.sendBeacon !== 'function') {
+      logger.warn('Beacon API not available, skipping sending outcomes.');
+      return;
+    }
+
+    const outcomes = this._outcomes;
```

Comment:
Why assign to another var? Every byte counts lol

Reply:
We want to process current outcomes, but let the rest being gathered as we do it. Updated to clear the dict instantly after caching it.